internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -334,13 +334,15 @@ func (c *FunctionComposer) Compose(ctx context.Context, xr *composite.Unstructur
 
 		// Pre-populate bootstrap requirements
 		if fn.Requirements != nil {
-			req.ExtraResources = map[string]*fnv1.Resources{}
+			// Bootstrap requirements were introduced alongside the new field names,
```

Comment:
it took me a second, but i think this is right - want to flag it just in case though 😊 

at first, i thought this was wrong, because this is not operations, this is composition, so I thought here we would need to support both field names. but this is specifically for "bootstrap" requirements, that was previously not possible with composition functions until the introduction of operations and the new field name. Is that right?

Reply:
Correct. This ability to specify requirements via the `spec.pipeline` of a Composition directly (as opposed to by returning them in a `RunFunctionRequest`) is new with this release. So I figured we should support the new way.